pkg/apiserver/registry/stats/nodelatencystats/rest.go

```diff
@@ -50,38 +57,112 @@ func (r *REST) Destroy() {
 func (r *REST) NewList() runtime.Object {
 	return &statsv1alpha1.NodeLatencyStats{}
 }
 
 func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
-	// TODO: fill this function in next PR
-	return &statsv1alpha1.NodeLatencyStatsList{}, nil
+	objs := r.indexer.List()
+
+	// Due to the unordered nature of map iteration and the complexity of controlling 'continue',
+	// we will ignore paging here and plan to implement it in the future.
+
+	entries := make([]statsv1alpha1.NodeLatencyStats, 0, len(objs))
+	for _, obj := range objs {
+		entries = append(entries, *obj.(*statsv1alpha1.NodeLatencyStats))
+	}
+
+	return &statsv1alpha1.NodeLatencyStatsList{Items: entries}, nil
 }
 
 func (r *REST) ConvertToTable(ctx context.Context, obj runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
-	// TODO: fill this function in next PR
 	table := &metav1.Table{
 		ColumnDefinitions: []metav1.TableColumnDefinition{
-			{Name: "SourceNodeName", Type: "string", Format: "name", Description: "Source node name."},
-			{Name: "NodeLatencyStatsList", Type: "array", Format: "string", Description: "Node IP latency list."},
+			{Name: "Node Name", Type: "string", Format: "name", Description: "Name of Node from which latency was measured."},
+			{Name: "Num Latency Entries", Type: "integer", Format: "int64", Description: "Number of peers for which latency measurements are available."},
+			{Name: "Max Latency", Type: "integer", Format: "int64", Description: "Name of the peer with the highest latency and the latency value(ns)."},
+			{Name: "Avg Latency", Type: "integer", Format: "int64", Description: "Average latency value across all peers(ns)."},
 		},
 	}
+	if m, err := meta.ListAccessor(obj); err == nil {
+		table.ResourceVersion = m.GetResourceVersion()
+		table.Continue = m.GetContinue()
+		table.RemainingItemCount = m.GetRemainingItemCount()
+	} else {
+		if m, err := meta.CommonAccessor(obj); err == nil {
+			table.ResourceVersion = m.GetResourceVersion()
+		}
+	}
 
-	return table, nil
+	var err error
+	table.Rows, err = metatable.MetaToTableRow(obj, func(obj runtime.Object, m metav1.Object, name, age string) ([]interface{}, error) {
+		summary := obj.(*statsv1alpha1.NodeLatencyStats)
+
+		// Calculate the max and average latency values.
+		peerNodeLatencyEntriesCount := len(summary.PeerNodeLatencyStats)
+		var targetIPLatencyCount int64
+		var maxLatency int64
+		var avgLatency int64
+
+		for i := range summary.PeerNodeLatencyStats {
+			targetIPLatency := summary.PeerNodeLatencyStats[i]
+
+			for j := range targetIPLatency.TargetIPLatencyStats {
+				targetIPLatencyCount++
+				currentLatency := targetIPLatency.TargetIPLatencyStats[j].LastMeasuredRTTNanoseconds
+				if currentLatency > maxLatency {
+					maxLatency = currentLatency
+				}
+
+				if targetIPLatencyCount == 1 {
+					avgLatency = currentLatency
+				} else {
+					avgLatency += (currentLatency - avgLatency) / targetIPLatencyCount
+				}
```

Comment:
you should do `avgLatency += currentLatency` here, and divide at the end to avoid rounding errors.
(not worried about an overflow given that this is an `int64`)

Reply:
Ok, I've changed it to simple accumulation.